sklearn/tests/test_pipeline.py

```diff
@@ -472,3 +561,50 @@ def test_X1d_inverse_transform():
     X = np.ones(10)
     msg = "1d X will not be reshaped in pipeline.inverse_transform"
     assert_warns_message(FutureWarning, msg, pipeline.inverse_transform, X)
+
+
+def test_set_feature_union_steps():
+    mult2 = MultT(2)
+    mult2.get_feature_names = lambda: ['x2']
+    mult3 = MultT(3)
+    mult3.get_feature_names = lambda: ['x3']
+    mult5 = MultT(5)
+    mult5.get_feature_names = lambda: ['x5']
+
+    ft = FeatureUnion([('m2', mult2), ('m3', mult3)])
+    assert_array_equal([[2, 3]], ft.transform(np.asarray([[1]])))
+    assert_equal(['m2__x2', 'm3__x3'], ft.get_feature_names())
+
+    # Directly setting attr
+    ft.transformer_list = [('m5', mult5)]
+    assert_array_equal([[5]], ft.transform(np.asarray([[1]])))
+    assert_equal(['m5__x5'], ft.get_feature_names())
+
+    # Using set_params
+    ft.set_params(transformer_list=[('mock', mult3)])
+    assert_array_equal([[3]], ft.transform(np.asarray([[1]])))
+    assert_equal(['mock__x3'], ft.get_feature_names())
+
+    # Using set_params to replace single step
+    ft.set_params(mock=mult5)
+    assert_array_equal([[5]], ft.transform(np.asarray([[1]])))
+    assert_equal(['mock__x5'], ft.get_feature_names())
+
+
+def test_set_feature_union_step_none():
+    mult2 = MultT(2)
+    mult2.get_feature_names = lambda: ['x2']
+    mult3 = MultT(3)
+    mult3.get_feature_names = lambda: ['x3']
+
+    ft = FeatureUnion([('m2', mult2), ('m3', mult3)])
+    assert_array_equal([[2, 3]], ft.transform(np.asarray([[1]])))
+    assert_equal(['m2__x2', 'm3__x3'], ft.get_feature_names())
+
+    ft.set_params(m2=None)
+    assert_array_equal([[3]], ft.transform(np.asarray([[1]])))
+    assert_equal(['m3__x3'], ft.get_feature_names())
+
+    ft.set_params(m3=None)
```

Comment:
It just seemed to be the logical extension. A grid search where every component is switched on and off, in which the `FeatureUnion` is further composed with other components?